README.md
## react-native-system-setting
It provides some system setting APIs for you. Support iOS and Android both.

### Support

* Volume ( with listener)
* Brightness
* Wifi switch
* Location
* Bluetooth
* Airplane

### Note

**[Example](https://github.com/c19354837/react-native-system-setting/tree/master/examples/SystemSettingExample) only work in the real device**

### Change Log

[Change Log](https://github.com/c19354837/react-native-system-setting/blob/master/CHANGELOG.md)

breaking change for permission since V1.5.0, see [Android Permission](https://github.com/c19354837/react-native-system-setting#android-permission)

## Look like

I really want to show the .gif, while it has no difference with .jpg for some system limit.

I strongly recommend you to run the example in real device to see how it works.

<img src="https://raw.githubusercontent.com/c19354837/react-native-system-setting/master/screenshot/ios.png" width = "40%"/>&nbsp;&nbsp;&nbsp;
<img src="https://raw.githubusercontent.com/c19354837/react-native-system-setting/master/screenshot/android.jpg" width = "40%" />

## Install
Run `npm i -S react-native-system-setting`

**Note: if your project was created by [Create React Native App](https://github.com/react-community/create-react-native-app), you should [Eject](https://github.com/react-community/create-react-native-app/blob/master/EJECTING.md) before link it.**

#### iOS
Run `react-native link` to link this library.

Or add `pod 'RCTSystemSetting', :path => '../node_modules/react-native-system-setting'` in `Podfile` for Cocoapods.

If link does not work, you can do it [manually](https://facebook.github.io/react-native/docs/linking-libraries-ios.html).

#### Android
Run `react-native link` to link this library.

That's all.

If link does not work, you can do it manually. Just follow this way:

**android/settings.gradle**

```gradle
include ':react-native-system-setting'
project(':react-native-system-setting').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-system-setting/android')
```

**android/app/build.gradle**

```gradle
dependencies {
   ...
   compile project(':react-native-system-setting')
}
```

**android/app/src/main/java/..packageName../MainApplication.java**

On top, where imports are:

```java
import com.ninty.system.setting.SystemSettingPackage;
```

Add the `SystemSettingPackage` class to your list of exported packages.

```java
@Override
protected List<ReactPackage> getPackages() {
    return Arrays.asList(
            new MainReactPackage(),
            new SystemSettingPackage()
    );
}
```

## Usage

**Common import**

```javascript
import SystemSetting from 'react-native-system-setting'
```

**volume**

```javascript
//get the current volume
SystemSetting.getVolume().then((volume)=>{
    console.log('Current volume is ' + volume);
});

// change the volume
SystemSetting.setVolume(0.5);

// listen only to some volume changes, based on the change reason (AVSystemController_AudioVolumeChangeReasonNotificationParameter). 
// If missing API call or sent an empty array it will listen to all changes. 
// Pass reasons like "ExplicitVolumeChange", "RouteChange", et al
SystemSetting.setVolumeChangeListenerReasons(["ExplicitVolumeChange"])

// listen the volume changing if you need
const volumeListener = SystemSetting.addVolumeListener((data) => {
    const volume = data.value;
    console.log(volume);
});

//remove listener when you need it no more
SystemSetting.removeVolumeListener(volumeListener)       
```

> `setVolume` can do more, [more detail](https://github.com/c19354837/react-native-system-setting/blob/master/API.md) 

**brightness**

```javascript
//get the current brightness
SystemSetting.getBrightness().then((brightness)=>{
    console.log('Current brightness is ' + brightness);
});

//change the brightness & check permission
SystemSetting.setBrightnessForce(0.5).then((success)=>{
    !success && Alert.alert('Permission Deny', 'You have no permission changing settings',[
	   {'text': 'Ok', style: 'cancel'},
	   {'text': 'Open Setting', onPress:()=>SystemSetting.grantWriteSettingPermission()}
	])
});

// save the value of brightness and screen mode.
SystemSetting.saveBrightness();
// restore the brightness and screen mode. you can get the old brightness value.
SystemSetting.restoreBrightness().then((oldVal)=>{
    //if you need
})

// change app's brightness without any permission.
SystemSetting.setAppBrightness(0.5);
SystemSetting.getAppBrightness().then((brightness)=>{
    console.log('Current app brightness is ' + brightness);
})
```

> `setBrightness()` & `saveBrightness()` need [permission](https://github.com/c19354837/react-native-system-setting#android-permission) for Android

**Wifi**

```javascript
SystemSetting.isWifiEnabled().then((enable)=>{
    const state = enable ? 'On' : 'Off';
    console.log('Current wifi is ' + state);
})

SystemSetting.switchWifi(()=>{
    console.log('switch wifi successfully');
})
```

> `isWifiEnabled()` need [permission](https://github.com/c19354837/react-native-system-setting#android-permission) for Android
> 
> `switchWifi()` is disabled by default for iOS since V1.7.0, [enable it](https://github.com/c19354837/react-native-system-setting/blob/master/iOS.md#ios)

**Location**

```javascript
SystemSetting.isLocationEnabled().then((enable)=>{
    const state = enable ? 'On' : 'Off';
    console.log('Current location is ' + state);
})

SystemSetting.switchLocation(()=>{
    console.log('switch location successfully');
})
```
> `switchLocation()` is disabled by default for iOS since V1.7.0, [enable it](https://github.com/c19354837/react-native-system-setting/blob/master/iOS.md#ios)

**Bluetooth**

```javascript
SystemSetting.isBluetoothEnabled().then((enable)=>{
    const state = enable ? 'On' : 'Off';
    console.log('Current bluetooth is ' + state);
})

SystemSetting.switchBluetooth(()=>{
    console.log('switch bluetooth successfully');
})
```

> `isBluetoothEnabled()` need [permission](https://github.com/c19354837/react-native-system-setting#android-permission) for Android
>
> All bluetooth-function are disabled by default for iOS since V1.7.0, [enable it](https://github.com/c19354837/react-native-system-setting/blob/master/iOS.md#ios)

**Airplane**

```javascript
SystemSetting.isAirplaneEnabled().then((enable)=>{
    const state = enable ? 'On' : 'Off';
    console.log('Current airplane is ' + state);
})

SystemSetting.switchAirplane(()=>{
    console.log('switch airplane successfully');
})
```

> `isAirplaneEnabled()` will always return `true` for iOS if your device has no SIM card, see [detail](https://github.com/c19354837/react-native-system-setting/issues/37)
> 
> `switchAirplane()` is disabled by default for iOS since V1.7.0, [enable it](https://github.com/c19354837/react-native-system-setting/blob/master/iOS.md#ios)

**Other**

```javascript
// open app setting page
SystemSetting.openAppSystemSettings()
```

## API

[API](https://github.com/c19354837/react-native-system-setting/blob/master/API.md)

## Run example

```
$ cd example/SystemSettingExample
$ npm install
// if android
$ react-native run-android
// else
$ react-native run-ios
```

## iOS

To be more friendly to app store, I disable some APIs for iOS since V1.7.0, You can [enable it](https://github.com/c19354837/react-native-system-setting/blob/master/iOS.md#ios) in a few steps.

## Android permission
 
### API

Default permissions are removed since V1.5.0, see [this PR](https://github.com/c19354837/react-native-system-setting/pull/44). You need to declare the corresponding permissions in your app's AndroidManifest.xml, see [example AndroidManifest.xml](https://github.com/c19354837/react-native-system-setting/blob/master/examples/SystemSettingExample/android/app/src/main/AndroidManifest.xml)

**`android/app/src/main/AndroidManifest.xml`**

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="YourPackageName"
    android:versionCode="1"
    android:versionName="1.0">
    
    <!-- setBrightness() & setScreenMode() & saveBrightness() -->
    <uses-permission android:name="android.permission.WRITE_SETTINGS" />
    
    <!-- isWifiEnabled() -->
    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
    
    <!-- isBluetoothEnabled() -->
    <uses-permission android:name="android.permission.BLUETOOTH"/>
    
    <!-- * switchWifiSilence() -->
    <uses-permission android:name="android.permission.CHANGE_WIFI_STATE"/>

    <!-- * switchBluetoothSilence() -->
    <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
    
    ...

</manifest>
```

> There are some different APIs that end with `silence`. They can do the job programmatically without direct user consent. These APIs maybe useful when you develop a system management app. Or, you should call `switchWifi()` & `switchBluetooth()` to get a better user experience

### Do Not Disturb

`setVolume()` may cause a crash: **Not allowed to change Do Not Disturb state**. See [detail](https://github.com/c19354837/react-native-system-setting/issues/48).

### Runtime permission for Android 6+

Change *brightness* and *screen mode* need `android.permission.WRITE_SETTINGS` which user can disable it in phone Setting. When you call `setScreenMode()`, `setBrightness()` or `setBrightnessForce()` , it will return false if the app has no permission, and you can call `SystemSetting.grantWriteSettingPermission()` to guide user to app setting page. see [example](https://github.com/c19354837/react-native-system-setting/tree/master/examples/SystemSettingExample)

> If you just want to change app's brightness, you can call `setAppBrightness(val)`, and it doesn't require any permission. see [API](https://github.com/c19354837/react-native-system-setting/blob/master/API.md)

## In the end

Feel free to open issue or pull request

## License
[**MIT**](https://github.com/c19354837/react-native-system-setting/blob/master/LICENSE.md)


CHANGELOG.md
# Unrealeased
Allow users to only listen to some volume change notification reasons
# V1.7.6
**2020-10-11**

fix TypeScript definitions

# V1.7.5
**2020-09-02**

adds TypeScript definitions, see [detail](https://github.com/c19354837/react-native-system-setting/pull/116)

# V1.7.4
**2020-01-30**

typo: `grantWriteSettingPremission()` -> `grantWriteSettingPermission`, see [detail](https://github.com/c19354837/react-native-system-setting/issues/97)

# V1.7.3
**2019-10-03**

fix bug: `setBrightness()` will cause a crash in iOS 13, see [detail](https://github.com/c19354837/react-native-system-setting/pull/89)

# V1.7.2
**2019-02-14**

new API: `addLocationModeListener()`, only works in Android

# V1.7.1
**2018-12-30**

new API: `openAppSystemSettings` will open app setting page

# V1.7.0
**2018-09-30**

add preprocessor `PRIVATE_API` which can enable `switch*()` function, see [detail](https://github.com/c19354837/react-native-system-setting/issues/58)

`setAppStore()` is deprecated.

breaking change: remove `BLUETOOTH` by default.

# V1.6.0
**2018-09-19**

bluetooth issues for App Store, see [detail](https://github.com/c19354837/react-native-system-setting/issues/55)

# V1.5.3
**2018-08-20**

fix bug: `setVolume()` may cause a crash, see [detail](https://github.com/c19354837/react-native-system-setting/pull/54)

# V1.5.2
**2018-08-05**

`setVolume()` may crash in >= Android M, see [detail](https://github.com/c19354837/react-native-system-setting/pull/48)

RN 0.56.+ compatible, [Use project-wide properties and new dependency](https://github.com/c19354837/react-native-system-setting/pull/46)

# V1.5.1
**2018-08-02**

show System Volume UI by default for iOS, see [detail](https://github.com/c19354837/react-native-system-setting/pull/43)

# V1.5.0
**2018-07-10**

breaking change: remove default Android permission, see [detail](https://github.com/c19354837/react-native-system-setting/pull/44)

# V1.4.6
**2018-06-20**

fix bug: `switchAirplane()` will open Bluetooth Setting Page for iOS.

# V1.4.5
**2018-06-07**

fix bug: Exception handling for Volume Event, see [detail](https://github.com/c19354837/react-native-system-setting/issues/39) 

# V1.4.4
**2018-05-27**

new API: `getLocationMode()`, see [detail](https://github.com/c19354837/react-native-system-setting/issues/35)

fix bug: all listener may get `null` sometime 

# V1.4.3
**2018-05-21**

for iOS - override `+(BOOL)requiresMainQueueSetup` to remove warning

# V1.4.2
**2018-05-07**

new API: `addLocationListener()`, only works in Android

new API: `addAirplaneListener()`, only works in Android

# V1.4.1
**2018-04-23**

new API: `addWifiListener()`, only works in Android

new API: `addBluetoothListener()`, Android & iOS

# V1.4.0
**2018-04-21**

new API: `setAppStore()`, it will avoid some troubles when you submit app to App Store, see [detail](https://github.com/c19354837/react-native-system-setting/issues/28)

# V1.3.0
**2018-04-15**

new API: `switchBluetoothSilence()`, see [detail](https://github.com/c19354837/react-native-system-setting/blob/master/API.md)

**breaking change**: remove `android.permission.CHANGE_WIFI_STATE` from [AndroidManifest.xml](https://github.com/c19354837/react-native-system-setting/blob/master/android/src/main/AndroidManifest.xml). If you want to call `switchWifiSilence()`, you should declare the permission in your Androidmanifest.xml, see [detail](https://github.com/c19354837/react-native-system-setting#powerful-api)

# V1.2.5
**2018-04-07**

fix bug - `setVolume(val, config)` will cause a crash when `type` is null, see [detail](https://github.com/c19354837/react-native-system-setting/issues/22)

fix bug - `grantWriteSettingPermission()` navigates to the wrong page, see [detail](https://github.com/c19354837/react-native-system-setting/issues/24)

# V1.2.4
**2018-03-14**

fix bug - `isLocationEnabled` return `false` when locate by network only, see [detail](https://github.com/c19354837/react-native-system-setting/issues/19)

# V1.2.3
**2018-02-26**

fix bug - `setVolume` will show the `MPVolumeView` in the upper left corner, see [detail](https://github.com/c19354837/react-native-system-setting/issues/17)

# V1.2.2
**2018-02-25**

use `MPVolumeView` to get/set volume for iOS.

change API `setVolume(val, type)` to `setVolume(val, config)`, see [detail](https://github.com/c19354837/react-native-system-setting/issues/15)

# V1.2.1
**2018-01-21**

Supported Cocoapods, see [detail](https://github.com/c19354837/react-native-system-setting/issues/10)

# V1.2.0
**2018-01-07**

new API: `isAirplaneEnabled()` & `switchAirplane(complete)`, see [detail](https://github.com/c19354837/react-native-system-setting/blob/master/API.md)

# V1.1.1
**2018-01-03**

Supported >= iOS 8.0.

new API: `setAppBrightness(val:float)` & `getAppBrightness()`, see [detail](https://github.com/c19354837/react-native-system-setting/blob/master/API.md)

# V1.1.0
**2017-12-14**

Check permission when write setting, see [detail](https://github.com/c19354837/react-native-system-setting#runtime-permission-for-android-6)

# V1.0.8
**2017-11-27**

Now you can change volume by type (Android only), see [detail](https://github.com/c19354837/react-native-system-setting/issues/5)

# V1.0.7
**2017-08-19**

support RN 0.47, see [detail](https://github.com/c19354837/react-native-system-setting/issues/1)

# V1.0.6
**2017-07-23**

support switching bluetooth state

# V1.0.5
**2017-07-15**

support switching location state

API change : `switchWifi()` and `switchWifiSilence()`, see [detail](https://github.com/c19354837/react-native-system-setting/blob/master/API.md)

# V1.0.4
**2017-06-16**

support switching wifi state

# V1.0.3
**2017-06-05**

Save & Restore for brightness

# V1.0.2
**2017-06-03**

Add listener for volume

# V1.0.1
**2017-05-31**

Add screen mode (Android only)

# V1.0.0
**2017-05-30**

Support volume and brightness


examples/SystemSettingExample/index.js
import React, { Component } from 'react'
import { AppRegistry, StyleSheet, Alert, Text, View, Slider, TouchableOpacity, PixelRatio, Switch, ActivityIndicator, ScrollView, Platform } from 'react-native'

import SystemSetting from 'react-native-system-setting'

export default class SystemSettingExample extends Component {

    isAndroid = Platform.OS === 'android'

    volumeListener = null
    wifiListener = null
    bluetoothListener = null
    locationListener = null
    airplaneListener = null

    volTypes = ['music', 'system', 'call', 'ring', 'alarm', 'notification']
    volIndex = 0

    constructor(props) {
        super(props)
        this.state = {
            volume: 0,
            volType: this.volTypes[this.volIndex],
            brightness: 0,
            wifiEnable: false,
            wifiStateLoading: false,
            locationEnable: false,
            locationStateLoading: false,
            bluetoothEnable: false,
            bluetoothStateLoading: false,
            airplaneEnable: false,
            airplaneStateLoading: false,
        }
    }

    async componentDidMount() {
        SystemSetting.setAppStore(false)
        this.setState({
            volume: await SystemSetting.getVolume(this.state.volType),
            brightness: await SystemSetting.getBrightness(),
            wifiEnable: await SystemSetting.isWifiEnabled(),
            locationEnable: await SystemSetting.isLocationEnabled(),
            bluetoothEnable: await SystemSetting.isBluetoothEnabled(),
            airplaneEnable: await SystemSetting.isAirplaneEnabled(),
        })
        // just init slider value directly
        this._changeSliderNativeVol(this.sliderVol, this.state.volume)
        this._changeSliderNativeVol(this.sliderBri, this.state.brightness)

        this.volumeListener = SystemSetting.addVolumeListener((data) => {
            const volume = this.isAndroid ? data[this.state.volType] : data.value
            this._changeSliderNativeVol(this.sliderVol, volume)
            this.setState({
                volume: volume
            })
        })

        this.wifiListener = await SystemSetting.addWifiListener((enable) => {
            this.setState({ wifiEnable: enable })
        })

        this.bluetoothListener = await SystemSetting.addBluetoothListener((enable) => {
            this.setState({ bluetoothEnable: enable })
        })

        this.locationListener = await SystemSetting.addLocationListener((enable) => {
            this.setState({ locationEnable: enable })
        })

        this.airplaneListener = await SystemSetting.addAirplaneListener((enable) => {
            this.setState({ airplaneEnable: enable })
        })
    }

    _changeSliderNativeVol(slider, value) {
        slider.setNativeProps({
            value: value
        })
    }

    componentWillUnmount() {
        SystemSetting.removeListener(this.volumeListener)
        SystemSetting.removeListener(this.wifiListener)
        SystemSetting.removeListener(this.bluetoothListener)
        SystemSetting.removeListener(this.locationListener)
        SystemSetting.removeListener(this.airplaneListener)
    }

    _changeVol(value) {
        SystemSetting.setVolume(value, {
            type: this.state.volType,
            playSound: false,
            showUI: false
        })
        this.setState({
            volume: value
        })
    }

    _changeVolType = async () => {
        this.volIndex = ++this.volIndex % this.volTypes.length
        const volType = this.volTypes[this.volIndex]
        const vol = await SystemSetting.getVolume(volType)
        this._changeSliderNativeVol(this.sliderVol, vol)
        this.setState({
            volType: volType,
            volume: vol
        })
    }

    _changeBrightness = async (value) => {
        const result = await SystemSetting.setBrightnessForce(value)
        if (!result) {
            Alert.alert('Permission Deny', 'You have no permission changing settings', [
                { 'text': 'Ok', style: 'cancel' },
                { 'text': 'Open Setting', onPress: () => SystemSetting.grantWriteSettingPremission() }
            ])
            return
        }
        this.setState({
            brightness: value,
        })
    }

    _restoreBrightness() {
        const saveBrightnessVal = SystemSetting.restoreBrightness()
        if (saveBrightnessVal > -1) {
            // success
            this.setState({
                brightness: saveBrightnessVal
            })
            this._changeSliderNativeVol(this.sliderBri, saveBrightnessVal)
        }
    }

    _switchWifi() {
        this.setState({
            wifiStateLoading: true
        })
        SystemSetting.switchWifi(async () => {
            this.setState({
                wifiEnable: await SystemSetting.isWifiEnabled(),
                wifiStateLoading: false
            })
        })
    }

    _switchLocation() {
        this.setState({
            locationStateLoading: true
        })
        SystemSetting.switchLocation(async () => {
            this.setState({
                locationEnable: await SystemSetting.isLocationEnabled(),
                locationStateLoading: false
            })
        })
    }

    _switchBluetooth() {
        this.setState({
            bluetoothStateLoading: true
        })
        SystemSetting.switchBluetooth(async () => {
            this.setState({
                bluetoothEnable: await SystemSetting.isBluetoothEnabled(),
                bluetoothStateLoading: false
            })
        })
    }

    _switchAirplane() {
        this.setState({
            airplaneStateLoading: true
        })
        SystemSetting.switchAirplane(async () => {
            this.setState({
                airplaneEnable: await SystemSetting.isAirplaneEnabled(),
                airplaneStateLoading: false
            })
        })
    }

    render() {
        const { volume, brightness,
            wifiEnable, wifiStateLoading,
            locationEnable, locationStateLoading,
            bluetoothEnable, bluetoothStateLoading,
            airplaneEnable, airplaneStateLoading,
        } = this.state
        return (
            <ScrollView style={styles.container}>
                <View style={styles.head}>
                </View>
                <ValueView
                    title='Volume'
                    btn={this.isAndroid && {
                        title: this.state.volType,
                        onPress: this._changeVolType
                    }}
                    value={volume}
                    changeVal={(val) => this._changeVol(val)}
                    refFunc={(sliderVol) => this.sliderVol = sliderVol}
                />
                <ValueView
                    title='Brightness'
                    value={brightness}
                    changeVal={(val) => this._changeBrightness(val)}
                    refFunc={(sliderBri) => this.sliderBri = sliderBri}
                />
                <View style={styles.card}>
                    <View style={styles.row}>
                        <Text style={styles.title}>Brightness save & restore
                        </Text>
                    </View>
                    <View style={styles.row}>
                        <TouchableOpacity style={{ marginRight: 32 }} onPress={SystemSetting.saveBrightness}>
                            <Text style={styles.btn}>Save
                            </Text>
                        </TouchableOpacity>
                        <TouchableOpacity onPress={this._restoreBrightness.bind(this)}>
                            <Text style={styles.btn}>Restore
                            </Text>
                        </TouchableOpacity>
                    </View>
                </View>
                <StatusView
                    title='Wifi'
                    value={wifiEnable}
                    loading={wifiStateLoading}
                    switchFunc={(value) => this._switchWifi()} />
                <StatusView
                    title='Location'
                    value={locationEnable}
                    loading={locationStateLoading}
                    switchFunc={(value) => this._switchLocation()} />
                <StatusView
                    title='Bluetooth'
                    value={bluetoothEnable}
                    loading={bluetoothStateLoading}
                    switchFunc={(value) => this._switchBluetooth()} />
                <StatusView
                    title='Airplane'
                    value={airplaneEnable}
                    loading={airplaneStateLoading}
                    switchFunc={(value) => this._switchAirplane()} />
            </ScrollView>
        )
    }
}

const ValueView = (props) => {
    const { title, value, changeVal, refFunc, btn } = props
    return (
        <View style={styles.card}>
            <View style={styles.row}>
                <Text style={styles.title}>{title}</Text>
                {btn && <TouchableOpacity onPress={btn.onPress}>
                    <Text style={styles.btn}>{btn.title}
                    </Text>
                </TouchableOpacity>}
                <Text style={styles.value}>{value}</Text>
            </View>
            <Slider
                ref={refFunc}
                style={styles.slider}
                onValueChange={changeVal} />
        </View>
    )
}

const StatusView = (props) => {
    const { title, switchFunc, value, loading } = props
    return (
        <View style={styles.card}>
            <View style={styles.row}>
                <Text style={styles.title}>{title}
                </Text>
            </View>
            <View style={styles.row}>
                <Text>Current status is {loading ? 'switching' : (value ? 'On' : 'Off')}
                </Text>
                {
                    loading && <ActivityIndicator animating={loading} />
                }
                <View style={{ flex: 1, alignItems: 'flex-end' }}>
                    <Switch
                        onValueChange={switchFunc}
                        value={value} />
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#E5E7E8'
    },
    head: {
        justifyContent: 'center',
        alignItems: 'center',
        height: 64,
    },
    headline: {
        fontSize: 22,
        color: '#666'
    },
    card: {
        padding: 8,
        backgroundColor: '#fff',
        marginTop: 4,
        marginBottom: 4,
    },
    row: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingTop: 8,
        paddingBottom: 8,
    },
    title: {
        fontSize: 16,
        color: '#6F6F6F'
    },
    value: {
        fontSize: 14,
        flex: 1,
        textAlign: 'right',
        color: '#904ED9'
    },
    split: {
        marginVertical: 16,
        height: 1,
        backgroundColor: '#ccc',
    },
    btn: {
        fontSize: 16,
        padding: 8,
        paddingVertical: 8,
        color: '#405EFF'
    }
})

AppRegistry.registerComponent('SystemSettingExample', () => SystemSettingExample)
